docs/source/developing/plugin.rst

```diff
@@ -346,47 +348,251 @@ needed to perform tasks such as listening on a network connection, making
+        SetInputRunner(ir InputRunner)
+        Splitter() Splitter
+        SplitBytes(data []byte, del Deliverer) error
+        SplitStream(r io.Reader, del Deliverer) error
+        GetRemainingData() (record []byte)
+        GetRecordFromStream(r io.Reader) (int, []byte, error)
+        DeliverRecord(record []byte, del Deliverer)
+        KeepTruncated() bool
+        UseMsgBytes() bool
+        SetPackDecorator(decorator func(*PipelinePack))
+    }
+
+Don't let this scare you, however. SplitterRunner's expose some internal
+workings to be able to support advanced uses, but in most cases you only need
+to deal with a few of the exposed methods. Specifically, you care about either
+``SplitStream`` or ``SplitBytes``, and possibly about ``SetPackDecorator`` and
+``UseMsgBytes``.
+
+First we'll examine the "Split" methods. As mentioned above, you'll typically
+only want to use one or the other. Deciding which you want is straightforward.
+If your mechanism for getting data from the outside world is a stream object
+(an `io.Reader`, in Go terms), then you'll want SplitStream. If not and you
+just end up with a byte slice of binary data, then you'll want SplitBytes.
+
+Note that both SplitStream and SplitBytes ask for a ``Deliverer`` object as
+their second argument. Again, in simple cases you don't need to worry about
+this. If you're only using a single SplitterRunner, you can just pass in nil
+and Heka will take care of delivering the message to a decoder and/or the
+message router appropriately. If you're using multiple goroutines (and
+therefore multiple SplitterRunners), however, you'll typically want multiple
+Deliverers, too. This is especially important if you want each separate
+goroutine to have its own Decoder, so decoding can happen in parallel,
+delegated to multiple cores on a single machine.
+
+Like SplitterRunners, Deliverers are obtained from the InputRunner, using the
+``NewDeliverer`` method. And, like SplitterRunners, NewDeliverer takes a
+single string identifier argument, which should be unique for each requested
+deliverer. Usually a single SplitterRunner will be using a single Deliverer,
+and the same token identifier will be used for each. You can see an example of
+this in the TcpInput's `handleConnection` code snippet a bit further down this
+page.
+
+If you're using SplitBytes, then you'll want to call it each time you have a
+new payload of data to process. It will return the number of bytes
+successfully consumed from the provided slice, and any relevant errors
+occurred while processing. It is up to the calling code to decide what to do
+in error cases, or when all of the data isn't consumed.
+
+If you're using SplitStream, then the SplitStream call will block for as long
+as it is consuming data. When data processing pauses or stops, SplitStream
+will exit and return control back to the input, returning either nil or any
+relevant errors. Typically if nil is returned, you'll want to again call
```

Comment:
Oh, sorry, that was supposed to be "again call SplitStream again again". ;)

Fixing, thanks.